src/commands/mapsheet-coverage/mapsheet.coverage.ts

```diff
@@ -144,7 +145,26 @@ export const commandMapSheetCoverage = command({
       captureArea.properties['description'] = collection.description;
       captureArea.properties['id'] = collection.id;
       captureArea.properties['license'] = collection.license;
-      captureArea.properties['providers'] = collection.providers;
+      if (collection.providers) {
+        /*
+         We can't loop over `StacRoles from 'stac-ts'`, so we need to keep this list in sync with that type
```

Comment:
we can reuse the type

```
const Roles: Record<StacRoles, string[]> = { producer: [], processor: [], licensor: [], host: []}
```

This style will lead to runtime errors though as the STAC could contain other roles (eg invalid or newer stac versions), for instance `providers: [ { roles: ['abc123'], name: 'foo' } ]` which would cause a runtime error here.

I would recommend using a set so we de-dupe providers and allow unknown roles?

```typescript
const providerByRoles = new Map<string, Set<string>>();
for (const f of collection.providers ?? []) {
  for (const role of f.roles ?? []) {
    const roleSet = providerByRoles.get(role) ?? new Set();
    roleSet.add(f.name);
    providerByRoles.set(role, roleSet);
  }
}
for (const [name, values] of providerByRoles) captureArea.properties[name] = [...values].join(', ');
```